pkg/policy/api/cidr.go

```diff
@@ -40,6 +41,19 @@ func (c *CIDR) MatchesAll() bool {
 	return false
 }
 
+// MatchesLocalNode determines whether the CIDR matches the local node IP.
+func (c *CIDR) MatchesLocalNode() bool {
```

Comment:
nit: I think this method should be a value receiver (and thus `MatchesAll()` as well).

Reply:
I'm guessing this is just following the existing examples which also appear to use pointer receivers, I'm guessing in the past we haven't paid so close attention to this. Should we update the existing cases too?

Reply:
I'd say so although I don't think it's extremely important. When the base type is simple (such as a string here), there's virtually no overhead of using a value receiver and thus I think it's better to use a value receive as this guarantees that the method has no side-effect on `CIDR`. See also https://golang.org/doc/faq#methods_on_values_or_pointers

Reply:
Filed https://github.com/cilium/cilium/issues/12558 to follow up.